pywemo/ssdp.py

```diff
@@ -336,16 +366,137 @@ def scan(
                     if len(entries) == max_entries:
                         return entries
     except socket.error:
-        logging.getLogger(__name__).exception(
-            "Socket error while discovering SSDP devices"
-        )
+        LOG.exception("Socket error while discovering SSDP devices")
     finally:
         for s in sockets:
             s.close()
 
     return entries
 
 
+class DiscoveryResponder:
+    """Inform Wemo devices of the pywemo virtual Wemo device.
+
+    The DiscoveryResponder informs Wemo devices of the /setup.xml URL for the
+    pywemo virtual Wemo device. The virtual device is used for receiving long
+    press actions from Wemo devices and is integrated into the
+    SubscriptionRegistry HTTP server.
+
+    Wemo devices are informed of the pywemo virtual Wemo device in two ways:
+
+    1. Wemo devices periodically send UPnP M-SEARCH discovery requests for the
+    to locate other devices on the network. DiscoveryResponder responds to these
+    requests with the URL for the virtual device.
+
+    2. A UPnP NOTIFY message is periodically multicasted by DiscoveryResponder
+    to inform Wemo devices on the network of the URL for the virtual device.
+    """
+
+    def __init__(self, callback_port: int):
+        """Create a server that will respond to WeMo discovery requests.
+
+        Args:
+            callback_port: The port for the SubscriptionRegistry HTTP server.
+        """
+        self.callback_port = callback_port
+        self._thread = None
+        self._exit = threading.Event()
+        self._thread_exception = None
+        self._notify_enabled = True  # For tests.
+
+    def send_notify(self) -> None:
+        """Send a UPnP NOTIFY message containing the virtual device URL."""
+        ssdp_target = (MULTICAST_GROUP, MULTICAST_PORT)
+        for addr in interface_addresses():  # Send on all interfaces.
+            callback_addr = (addr, self.callback_port)
+            sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            try:
+                sock.bind((addr, 0))
+                sock.sendto(
+                    (SSDP_NOTIFY % callback_addr).encode("UTF-8"), ssdp_target
+                )
+            except socket.error:
+                pass
+            finally:
+                sock.close()
```

Comment:
I have done very little python work with sockets, is there a difference between `sock.close` and deleting it with `del sock`, which is used in the `get_ip_address` function?  Some examples I've seen online call `sock.close()` followed by `del sock`.  It seems like the variable goes out of scope immediately anyway, but I've heard of occasional objects that need an explicit `del` at times.